### PR TITLE
feat: add components to rails stats task

### PR DESCRIPTION
**Pull request by nicolas-brousse**
### Summary

Add `app/components` to `rails stats` task.

### Other Information

Inspiration from https://github.com/rspec/rspec-rails/blob/main/lib/rspec/rails/tasks/rspec.rake.

Not really sure what could I add for test coverage.

### CODE REVIEW

File: lib/view_component/rails/tasks/view_component.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+task stats: "view_component:statsetup"
+
+namespace :view_component do
+  task statsetup: :environment do
+    require "rails/code_statistics"
+
+    ::STATS_DIRECTORIES << ["Components", Rails.application.config.view_component.view_component_path]
```
Comment (joelhawksley) [inline lib/view_component/rails/tasks/view_component.rb:9]: What do you think about:

```suggestion
    ::STATS_DIRECTORIES << ["ViewComponents", Rails.application.config.view_component.view_component_path]
```

File: lib/view_component/engine.rb [hunk 2]
```diff
@@ -8,6 +8,10 @@ class Engine < Rails::Engine # :nodoc:
     config.view_component = ActiveSupport::OrderedOptions.new
     config.view_component.preview_paths ||= []
 
+    rake_tasks do
+      load "view_component/rails/tasks/view_component.rake"
```
Comment (joelhawksley) [inline lib/view_component/engine.rb:12]: Perhaps we could add a CI task that runs `rails stats`?

Comment (nicolas-brousse) [reply lib/view_component/engine.rb:12]: I took some time to search and I found a way to add a test for that task :)

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -23,6 +23,10 @@ title: Changelog
 
     *Dany Marcoux*
 
+* Add components to `rails stats` task.
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:26]: I believe this PR should be rebased to add this above the release I just shipped today ❤️

**Review by joelhawksley — APPROVED** [review]
One small note on rebasing/changelog. Will merge once that's addressed!

**Review by Spone — APPROVED** [review]

File: docs/CHANGELOG.md (hunk 3 above)
Comment (nicolas-brousse) [reply docs/CHANGELOG.md]: Done 🙂

**Pull request merged**